server/matchmaker/map_pool.py

```diff
@@ -0,0 +1,41 @@
+import random
+from collections import Counter
+from typing import List
+
+from ..decorators import with_logger
+from ..types import MapDescription
+
+
+@with_logger
+class MapPool(object):
+    def __init__(self, name: str, maps: List[MapDescription] = []):
+        self.name = name
+        self.set_maps(maps)
+
+    def set_maps(self, maps: List[MapDescription]) -> None:
+        self.maps = {map_.id: map_ for map_ in maps}
+        self.map_ids = set(self.maps.keys())
+
+    def choose_map(self, played_map_ids: List[int] = []) -> MapDescription:
```

Comment:
Maybe the logic would actually be simpler if this always picked (one of the) least played maps?

Reply:
True